src/components/PopupWithForm.js

```diff
@@ -8,20 +8,31 @@ export default class PopupWithForm extends Popup {
       this._popupForm.querySelectorAll(".modal__input")
     );
     this._handleFormSubmit = handleFormSubmit;
+    this._modalSubmitButton =
+      this._popupElement.querySelector(".modal__button");
   }
 
-  _setInputValues() {
-    this._formValues = {};
+  _getInputValues() {
+    const formValues = {};
     this._inputValues.forEach((input) => {
-      this._formValues[input.name] = input.value;
+      formValues[input.name] = input.value;
     });
-    return this._formValues;
+    return formValues;
   }
+
+  renderLoading(isLoading) {
+    if (isLoading) {
+      this._modalSubmitButton.textContent = "Saving...";
+    } else {
+      this._modalSubmitButton.textContent = "Save";
```

Comment:
**Could be improved**

It’s better to make method `renderLoading` inside class `PopupWithForm`. You should find the button once in the constructor and change the text:  

```js
  constructor({ popupSelector, handleSubmit }) {
    super(popupSelector);
    // fix the initial button text only once in the constructor
    this._submitBtnText = this._submitBtn.textContent
  }
  // add 2 params: isLoading and loadingText with a default text
  renderLoading(isLoading, loadingText='Saving...') {
    if (isLoading) {
      this._submitBtn.textContent = loadingText;
    } else {
	// here we return back the initial text. So, you don’t need to bother yourself about it
      this._submitBtn.textContent = this._submitBtnText;
    }
  }

```